kiali.go

```diff
@@ -90,6 +90,37 @@ func main() {
 	status.Put(status.CoreCommitHash, commitHash)
 	status.Put(status.ContainerVersion, determineContainerVersion(version))
 
+	// check kiali version compatibility with istio
+	meshName, meshVersion, err := status.GetMeshInfo()
+
+	ok := false
+	if err != nil {
+		log.Warningf("Failed to get mesh version, please check if url_service_version is configured correctly.")
+		status.Put(status.IsCompatible, "false")
+		status.AddWarningMessages("Failed to get mesh version, please check if url_service_version is configured correctly.")
+	} else if meshName == "Unknown" {
+		log.Warningf("Unknown Istio Implementation")
+		status.Put(status.MeshVersion, meshVersion)
+		status.Put(status.MeshName, meshName)
+		status.AddWarningMessages("Unknown Istio Implementation")
+	} else if !strings.Contains(meshName, "Istio") {
+		log.Info("Mesh Name: %v Mesh Version: %v", meshName, meshVersion)
+		status.Put(status.MeshVersion, meshVersion)
+		status.Put(status.MeshName, meshName)
+	} else {
+		ok = status.CheckMeshVersion(meshName, meshVersion, version)
+		if ok {
+			log.Info("Mesh Name: %v Mesh Version: %v", meshName, meshVersion)
+			status.Put(status.MeshVersion, meshVersion)
+			status.Put(status.MeshName, meshName)
+			status.Put(status.IsCompatible, "true")
+		} else {
+			log.Warningf("Kiali %v is not compatible with %v %v", version, meshName, meshVersion)
+			status.AddWarningMessages(fmt.Sprintf("Kiali %v is not compatible with %v %v", version, meshName, meshVersion))
+			status.Put(status.IsCompatible, "false")
+		}
+	}
```

Comment:
I haven't run this to confirm, but I strongly suspect this will emit warnings about incompatibility when Kiali is run in an OpenShift Service Mesh environment, which we do not want. We should enhance this feature being added in this PR by adding to the compatibility matrix all the valid Kiali <-> OSSM versions.

Reply:
Sure, when doing this task, I think we can check OSSM version at the same time as well.